File: ChessOpenings/ChessOpenings/Data/Game.java
import java.util.*;

public class Game
{
	private List<String> moves = new ArrayList<String>();
	private float result; //0 = black, 0.5 = draw 1 = white 
	private String gameString;
	
	public Game(String gameString)
	{
		this.gameString = gameString;
		if (gameString.contains("1/2-1/2"))
		{
			result = 0.5f;
		}
		else if (gameString.contains("1-0"))
		{
			result = 1;
		}
		else if (gameString.contains("0-1"))
		{
			result = 0;
		}
		else
		{
			result = -1;
		}
		extractMoves();
	}
	
	public float getResult()
	{
		return result;
	}

	public String toString()
	{
		String winningString = "";
		if (result == 1) winningString = "White wins";
		if (result == 0.5) winningString = "Draw";
		if (result == 0) winningString = "Black wins";
		String rtn = moves.size() + " moves - " + winningString;
		return rtn;
	}

	public boolean employsOpening(List<String> openingSequence)
	{
		boolean rtn = true;
		for (int i = 0; i < openingSequence.size(); i++)
		{
			if (i < moves.size())
			{
				if (!openingSequence.get(i).equals(moves.get(i)))
				{
					rtn = false;
				}
			}
			else
			{
				return false;
			}
		}
		//System.out.println("match found = " + rtn);
		return rtn;
	}

	private void extractMoves()
	{
		String strippedString = gameString.replaceAll("\\[.*\\]", "");
		//strippedString = strippedString.replaceAll("{.*}", "");
		strippedString = strippedString.replaceAll("\\$\\d", "");
		strippedString = strippedString.replaceAll("\\d*\\.", "");
		strippedString = strippedString.replaceAll("1/2-1/2", "");
		strippedString = strippedString.replaceAll("1-0", "");
		strippedString = strippedString.replaceAll("0-1", "");

		//System.out.println("string to split " + strippedString);

		String[] movesArray = strippedString.split("\\s");
		//System.out.println(movesArray.length);
		for (int i = 0; i < movesArray.length; i++)
		{
			String currentString = movesArray[i];
			if (currentString.trim().length() != 0)
			{
				moves.add(currentString);
				//System.out.println("Adding move - " + currentString);
			}
		}
	}	
}

File: ChessOpenings.DataConversion/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace ChessOpenings.DataConversion
{
    public class Game
    {
        private List<string> moves = new List<string>();
        public float Result { get; set; }
        private string gameString;

        public Game(string gameString)
        {
            this.gameString = gameString;
            if (gameString.Contains("1/2-1/2"))
            {
                Result = 0.5f;
            }
            else if (gameString.Contains("1-0"))
            {
                Result = 1;
            }
            else if (gameString.Contains("0-1"))
            {
                Result = 0;
            }
            else
            {
                Result = -1;
            }
            ExtractMoves();
        }

        public override string ToString()
        {
            string winningString = "";
            if (Result == 1) winningString = "White wins";
            if (Result == 0.5) winningString = "Draw";
            if (Result == 0) winningString = "Black wins";
            string rtn = moves.Count() + " moves - " + winningString;
            return rtn;
        }

        public bool EmploysOpening(List<string> openingSequence)
        {
            bool rtn = true;
            for (int i = 0; i < openingSequence.Count(); i++)
            {
                if (i < moves.Count())
                {
                    if (!openingSequence[i].Equals(moves[i]))
                    {
                        rtn = false;
                    }
                }
                else
                {
                    return false;
                }
            }
            return rtn;
        }

        private void ExtractMoves()
        {
            string strippedString = Regex.Replace(gameString, "\\[.*\\]", "");
            strippedString = Regex.Replace(strippedString, "\\$\\d", "");
            strippedString = Regex.Replace(strippedString, "\\d*\\.", "");
            strippedString = Regex.Replace(strippedString, "1/2-1/2", "");
            strippedString = Regex.Replace(strippedString, "1-0", "");
            strippedString = Regex.Replace(strippedString, "0-1", "");

            string[] movesArray = Regex.Split(strippedString, "\\s");

            for (int i = 0; i < movesArray.Length; i++)
            {
                string currentString = movesArray[i];
                if (currentString.Trim().Length != 0)
                {
                    moves.Add(currentString);
                }
            }
        }
    }
}
